client/signup/steps/woocommerce-install/confirm/index.tsx

```diff
@@ -151,8 +154,10 @@ export default function Confirm( props: WooCommerceInstallProps ): ReactElement
 						<StyledNextButton
 							disabled={ hasBlockers || ! isDataReady }
 							onClick={ () => {
+								dispatch( submitSignupStep( { stepName: 'confirm' }, { siteConfirmed: siteId } ) );
 								if ( siteUpgrading.required ) {
-									return ( window.location.href = siteUpgrading.checkoutUrl );
+									page( siteUpgrading.checkoutUrl );
```

Comment:
page uses a `setTimeout( () =>{}, 0);` which seems to solve the race condition for dispatch not being stored before the redirect happens.

Reply:
Ah, I've added that myself in the past when using `window.location.href`. Good to know that `page` handles it for us.